src/contexts/CartContext.tsx

```diff
@@ -1,16 +1,239 @@
+    const cachedCartProducts = this.getLocalCartProducts()
+    const cartProductOptions = this._restructureCachedCartProducts(cachedCartProducts)
+
+    const remoteCartProducts = await this._fetchRemoteCartProducts(operation, cachedCartProducts)
+    const mergedCartProducts = this._mergeLocalAndRemoteCartProducts({
+      remoteCartProducts,
+      cachedCartProducts,
+      cartProductOptions,
+    })
+    const availableProducts = this._removePhaseOutCartProducts(mergedCartProducts)
+
+    this._updateLocalCache(availableProducts)
+
+    if (!this.currentMemberId) {
+      return
+    }
+
+    try {
+      await this.updateCartProducts({
+        variables: {
+          memberId: this.currentMemberId,
+          cartProductObjects: availableProducts.map(product => {
+            const tracking = product?.options?.tracking || {}
+            return {
+              app_id: this.appId,
+              member_id: this.currentMemberId,
+              product_id: product.productId,
+              options: { tracking },
+            }
+          }),
+        },
+      })
+
+      return availableProducts
+    } catch (error) {}
+  }
+
+  public getLocalCartProducts(): CartProductProps[] {
+    let cachedCartProducts: CartProductProps[]
+    try {
+      localStorage.removeItem('kolable.cart')
+      localStorage.removeItem('kolable.cart.products')
+      cachedCartProducts = JSON.parse(localStorage.getItem('kolable.cart._products') || '[]')
+    } catch (error) {
+      cachedCartProducts = []
+    }
+    return cachedCartProducts
+  }
+
+  private _restructureCachedCartProducts(cachedCartProducts: CartProductProps[]): { [ProductId: string]: any } {
+    return cachedCartProducts.reduce((options, cartProduct) => {
+      options[cartProduct.productId] = cartProduct.options
+      return options
+    }, {} as { [ProductId: string]: any })
+  }
+
+  private _createGetCartProductOperationQuery(operation: cartOperation): string {
+    const productIdsCondition =
+      operation === cartOperation.REMOVE_CART_PRODUCTS
+        ? `product: { id: { _in: $productIds }, product_owner: { member: { app_id: { _eq: $appId } } } }`
+        : ''
+
+    return `
+      query GET_CART_PRODUCT_COLLECTION(
+        $appId: String!
+        $memberId: String!
+        $productIds: [String!]
+        $localProductIds: [String!]!
+        $merchandiseSpecIds: [uuid!]!
+      ) {
+        cart_product(
+          where: {
+            app_id: { _eq: $appId }
+            member_id: { _eq: $memberId }
+            ${productIdsCondition}
+          }
+        ) {
+          id
+          product {
+            id
+            type
+            product_owner {
+              member_id
+            }
+            product_enrollments(where: { member_id: { _eq: $memberId } }) {
+              member_id
+              is_physical
+            }
+          }
+        }
+        product(where: { id: { _in: $localProductIds } }) {
```

Comment:
I suggest putting all cart operations in this class, because other operator functions are scattered, and it makes the code messy.